kubernetes-aws-python/__main__.py

```diff
@@ -0,0 +1,34 @@
+import pulumi
+import pulumi_awsx as awsx
+import pulumi_eks as eks
+
+config = pulumi.Config()
+min_cluster_size = config.get_float("minClusterSize")
+if min_cluster_size is None:
+    min_cluster_size = 3
+max_cluster_size = config.get_float("maxClusterSize")
+if max_cluster_size is None:
+    max_cluster_size = 6
+desired_cluster_size = config.get_float("desiredClusterSize")
+if desired_cluster_size is None:
+    desired_cluster_size = 3
+eks_node_instance_type = config.get("eksNodeInstanceType")
+if eks_node_instance_type is None:
+    eks_node_instance_type = "t2.medium"
+vpc_network_cidr = config.get("vpcNetworkCidr")
+if vpc_network_cidr is None:
+    vpc_network_cidr = "10.0.0.0/16"
+eks_vpc = awsx.ec2.Vpc("eks-vpc",
+    enable_dns_hostnames=True,
+    cidr_block=vpc_network_cidr)
+eks_cluster = eks.Cluster("eks-cluster",
+    vpc_id=eks_vpc.vpc_id,
+    public_subnet_ids=eks_vpc.public_subnet_ids,
+    private_subnet_ids=eks_vpc.private_subnet_ids,
+    instance_type=eks_node_instance_type,
+    desired_capacity=desired_cluster_size,
+    min_size=min_cluster_size,
+    max_size=max_cluster_size,
+    node_associate_public_ip_address=False)
+pulumi.export("kubeconfig", eks_cluster.kubeconfig)
+pulumi.export("vpcId", eks_vpc.vpc_id)
```

Comment:
Here's how I'd do it idiomatically.

```suggestion
import pulumi
import pulumi_awsx as awsx
import pulumi_eks as eks

config = pulumi.Config()

min_cluster_size = 3 if config.get_float("minClusterSize") is None else config.get_float("minClusterSize")
max_cluster_size = 6 if config.get_float("maxClusterSize") is None else config.get_float("maxClusterSize")
desired_cluster_size = 3 if config.get_float("desiredClusterSize") is None else config.get_float("desiredClusterSize")
eks_node_instance_type = "t2.Medium" if config.get("eksNodeInstanceType") is None else config.get("eksNodeInstanceType")
vpc_network_cidr = "10.0.0.0/16" if config.get("vpcNetworkCidr") is None else config.get("vpcNetworkCidr")

eks_vpc = awsx.ec2.Vpc(
    "eks-vpc",
    enable_dns_hostnames=True,
    cidr_block=vpc_network_cidr
)

eks_cluster = eks.Cluster(
    "eks-cluster",
    vpc_id=eks_vpc.vpc_id,
    public_subnet_ids=eks_vpc.public_subnet_ids,
    private_subnet_ids=eks_vpc.private_subnet_ids,
    instance_type=eks_node_instance_type,
    desired_capacity=desired_cluster_size,
    min_size=min_cluster_size,
    max_size=max_cluster_size,
    node_associate_public_ip_address=False
)

pulumi.export("kubeconfig", eks_cluster.kubeconfig)
pulumi.export("vpcId", eks_vpc.vpc_id)
```

Reply:
Also, I notice comments in the YAML version and the Go version, but not the other language versions. Would they be helpful?

Reply:
Yes, comments would be useful, I think. I'll add them.

To be sure I'm interpreting your idiomatic code correctly, is this what you're suggesting?
```
min_cluster_size = 3 if config.get_float("minClusterSize") is None else config.get_float("minClusterSize")
```
(all on one line, no line breaks)

Reply:
Changed in 73aaab4

Reply:
> To be sure I'm interpreting your idiomatic code correctly, is this what you're suggesting?
> ```
> min_cluster_size = 3 if config.get_float("minClusterSize") is None else config.get_float("minClusterSize")
> ```
> (all on one line, no line breaks)

Yes, that's the main change to make it more idiomatic. I'm also adding some line breaks and reformatting the instance creation to make it a bit cleaner and match a few different Python style guides.

Reply:
@nimbinatus Your comment should be addressed, then; let me know if it isn't!